tensorboard/plugins/graph/tf_graph_common/hierarchy.ts

```diff
@@ -501,6 +550,37 @@ function addNodes(h: Hierarchy, graph: SlimGraph) {
         parent.deviceHistogram[node.device] =
             (parent.deviceHistogram[node.device] || 0) + 1;
       }
+
+      // Increment parents appropriate compatibility count
+      if (node.compatible) {
+        parent.compatibilityHistogram.compatible =
+            (parent.compatibilityHistogram.compatible || 0) + 1;
+      } else {
+        parent.compatibilityHistogram.incompatible =
+            (parent.compatibilityHistogram.incompatible || 0) + 1;
+      }
+
+      // Increment capability counts for in and out embeddings
+      _.each(node.inEmbeddings, (inNode) => {
+        if (inNode.compatible) {
+          parent.compatibilityHistogram.compatible =
+              (parent.compatibilityHistogram.compatible || 0) + 1;
+        } else {
+          parent.compatibilityHistogram.incompatible =
```

Comment:
I wonder if it's possible to encapsulate this logic within a function within this file, ie 

```
updateCompatibilityHistogram(node: Node) {
  if (node.compatible) {
  } else {
  }
}
```